### PR TITLE
[MIG] stock_reception_screen: Migration to 14.0

### PR DESCRIPTION
**Pull request by dzungtran89**

### CODE REVIEW

File: stock_reception_screen/models/stock_reception_screen.py [hunk 1]
```diff
@@ -500,13 +503,16 @@ def process_select_product(self):
 
     def _split_move(self, move, qty):
         # Hook intended to be overridden
-        return move._split(qty)
+        new_move_vals = move._split(qty)
+        if new_move_vals:
+            new_move = self.env["stock.move"].create(new_move_vals)
+            new_move._action_confirm()
```
Comment (sebalix) [inline stock_reception_screen/models/stock_reception_screen.py:509]: `_split_move` has to return a recordset:
```python
            [...]
            new_move._action_confirm()
            return new_move
        return move
```

File: stock_reception_screen/models/stock_reception_screen.py [hunk 2]
```diff
@@ -609,21 +615,23 @@ def process_set_expiry_date(self):
...
             )
             self.warn_notification = _(
                 "You cannot set a date prior to previous one ({})".format(
                     previous_life_date_str
                 )
             )
             return
-        self.current_move_line_lot_id.life_date = self.current_move_line_lot_life_date
+        self.current_move_line_lot_id.expiration_date = (
+            self.current_move_line_lot_life_date
```
Comment (sebalix) [inline stock_reception_screen/models/stock_reception_screen.py]: I would rename all `life_date` terms to `expiration_date` to be inline with the standard (this implies a pre-migration script to rename the column).

**Review by sebalix — COMMENTED** [review]
@dzungtran89 the module is not installable, can you check?

File: stock_reception_screen/models/stock_reception_screen.py (hunk 2 above)
Comment (dzungtran89) [reply stock_reception_screen/models/stock_reception_screen.py]: Hi @sebalix , I'll update the PR

File: stock_reception_screen/__manifest__.py [hunk 3]
```diff
@@ -27,6 +27,6 @@
         "views/stock_reception_screen.xml",
         "views/manual_barcode.xml",
     ],
-    "installable": True,
+    "installable": False,
```
Comment (sebalix) [inline stock_reception_screen/__manifest__.py]: @dzungtran89 sorry my mistake... the module was already flagged `installable` in the `pre-commit` commit, I was reviewing only your migration commit and didn't see the change.

File: stock_reception_screen/models/stock_reception_screen.py (hunk 1 above)
Comment (sebalix) [reply stock_reception_screen/models/stock_reception_screen.py:509]: You forgot to return `new_move` in the `if` block (or does GH rendering is tricking me?)

Comment (dzungtran89) [reply stock_reception_screen/models/stock_reception_screen.py:509]: it's mybad actually

File: stock_reception_screen/__manifest__.py (hunk 3 above)
Comment (dzungtran89) [reply stock_reception_screen/__manifest__.py]: @sebalix okay, I missed this comment too :(. Just pushed the PR

**Review by sebalix — APPROVED** [review]
LG! Thank you for all the forward ports.

**Review by hparfr — APPROVED** [review]

**Pull request merged**